src/endpoint/s3/s3_utils.js

```diff
@@ -668,6 +671,109 @@ function response_field_encoder_url(value) {
     return new URLSearchParams({ 'a': value }).toString().slice(2); // slice the leading 'a='
 }
 
+class S3SelectStream extends Transform {
+
+    static prelude_length_bytes = 12;
+    static crc_length_bytes = 4;
+
+    //end message is always the same, so we allocate and init it once.
+    static end_message = Buffer.concat([
+    Buffer.from([0, 0, 0, 56]), //PRELUDE begins: total byte length
+    Buffer.from([0, 0, 0, 40]), //headers byte length
+    Buffer.from([0xc1, 0xc6, 0x84, 0xd4]), //crc of above (end of PRELUDE)
+    Buffer.from([11]), //DATA begins, HEADERS begin: byte length of below header name
+    Buffer.from(':event-type'),
+    Buffer.from([7, 0, 3]), //type (7, string) of below header value
+    Buffer.from('End'),
+    Buffer.from([13]), //again for second header
+    Buffer.from(':message-type'),
+    Buffer.from([7, 0, 5]),
+    Buffer.from('event'), //end of HEADERS
+    //no PAYLOAD (DATA ends)
+    Buffer.from([0xfe, 0x2c, 0xee, 0x99]) //MESSAGE CRC (crc of above)
+    ]);
+
+//same for headers of records message
+    static records_message_headers = Buffer.concat([
+    Buffer.from([11]), //byte length of below header name
+    Buffer.from(':event-type'),
+    Buffer.from([7, 0, 7]), //type (7, string) of below header value
+    Buffer.from('Records'),
+    Buffer.from([13]), //again for content-type header
+    Buffer.from(':content-type'),
+    Buffer.from([7, 0, 24]),
+    Buffer.from('application/octet-stream'),
+    Buffer.from([13]), //again for message-type header
+    Buffer.from(':message-type'),
+    Buffer.from([7, 0, 5]),
+    Buffer.from('event')
+    ]);
+
+    constructor(opts) {
+        super(opts);
+        this.s3select = new (nb_native().S3Select)(opts);
+        this.http_res = opts.http_res;
+    }
+
+    async encode_chunk(select_result) {
+        dbg.log2("select res = ", select_result.select.toString());
+
+        const total_length_bytes =
+            select_result.select.length +
+            S3SelectStream.records_message_headers.length +
+            S3SelectStream.prelude_length_bytes +
+            S3SelectStream.crc_length_bytes;
+        const buffer = Buffer.alloc(S3SelectStream.prelude_length_bytes);
+        let index = 0;
+
+        //prelude:
+        index = buffer.writeUInt32BE(total_length_bytes);
+        index = buffer.writeUInt32BE(S3SelectStream.records_message_headers.length, index);
+        buffer.writeInt32BE(select_result.prelude_crc, index);
+        this.push(buffer);
+        //headers:
+        this.push(S3SelectStream.records_message_headers);
+        //body:
+        this.push(select_result.select);
+        //message crc:
+        const buffer2 = Buffer.alloc(S3SelectStream.crc_length_bytes);
+        buffer2.writeInt32BE(select_result.message_crc);
+        this.push(buffer2);
+
+        //dbg.log2("select content response = ", buffer.toString('hex').match(/../g).join(' '));
```

Comment:
@alphaprinz, please don't leave lines of code in the comments.
If you think it can be useful to us you can change the debug level to higher.

Reply:
I think it's useful for someone trying to debug this code.

Reply:
Great, we can uncomment it and set a high debug level in advance.

Reply:
maybe add a comment before it, for example:
```
// Uncomment this transforms the buffer into a human-readable string.
// it is not at any debug level since of heavy CPU costs and will be useful only in specific debug scenarios
```